OpenSim/Common/XsensDataReader.cpp

```diff
@@ -51,51 +52,46 @@ XsensDataReader::extendRead(const std::string& folderName) const {
 
         // Skip lines to get to data
         std::string line;
-        packetCounterIndex = -1; // Force moving file pointer to beginning of data for each stream
-        for (int j = 0; packetCounterIndex == -1; j++) {
-            std::getline(*nextStream, line);
-            if (j == 1 && SimTK::isNaN(dataRate)) { // Extract Data rate from line 1
-                std::vector<std::string> tokens = FileAdapter::tokenize(line, ", ");
-                // find Update Rate: and parse into dataRate
-                if (tokens.size() < 4) continue;
-                if (tokens[1] == "Update" && tokens[2] == "Rate:") {
-                    dataRate = std::stod(tokens[3]);
-                }
+        auto commentLine = true;
+        std::getline(*nextStream, line);
+        auto isCommentLine = [](std::string aline) {
+            return aline.substr(0, 2) == "//";
+        };
+        std::vector<std::string> tokens;
+        while (commentLine) {
+            // Comment lines of arbitrary naumber on the form // "key":"value"
```

Comment:
Typo in this line (I think should be "Comment lines of arbitrary number in the form")

And then spacing in next line: `tokens = FileAdapter::tokenize(line.substr(2), ":");`

Reply:
👍